City-of-Helsinki/kukkuu-ui
# KK-638, KK-639, KK-640, KK-641, KK-649 | Add support for event groups

## Description

In this PR I introduce the changes that are required for adding support for event groups. 

Apologies for lumping all the changes into the same PR. I felt a bit pressured with time and begun to develop the tasks before the API was ready--it was difficult to untangle the changes afterwards.

Added  
- Event group page
    - I added new layout logic which allows views to more easily support consistent loading and error behaviours

Changed  
- Refactored child event list component and modified it so that it supports event groups
- Improved test coverage for event view

Fixed  
- Missing background colors from buttons on hover

## Context




[KK-638](https://helsinkisolutionoffice.atlassian.net/browse/KK-638)
[KK-639](https://helsinkisolutionoffice.atlassian.net/browse/KK-639)
[KK-640](https://helsinkisolutionoffice.atlassian.net/browse/KK-640)
[KK-641](https://helsinkisolutionoffice.atlassian.net/browse/KK-641)
[KK-649](https://helsinkisolutionoffice.atlassian.net/browse/KK-649)

## How Has This Been Tested?

I've added some unit/integration tests for the new components and a browser test that checks that event groups can be used to access events.

## Manual Testing Instructions for Reviewers

As a logged in user

1. Select child
1. Expect to see event groups in the invitation list
1. Select event group
1. Expect to see a graphic, title, description and a list of events
1. Select event
1. Expect to see event details and a list of occurrences
1. Press back arrow within the UI x2
1. Expect to be back in your child's profile
